### PR TITLE
Refactor reusable workflow consistency verification + add e2e testing into PR CI

### LINKED ISSUE #246: End to end testing on PR submission
**Description**

Can run a workflow similar to the hourly workflow that verifies a checkpoint and expects output to disk.

### PR DESCRIPTION
**Pull request by linus-sun**
#### Summary

This PR refactors out portions of the rekor-monitor workflow--namely the consistency check component--into separate methods to improve readability of the primary workflow. These methods are currently being tested via the end-to-end test implemented [in this PR](https://github.com/sigstore/rekor-monitor/pull/472), and thus this PR depends on the aforementioned one. Future PRs will implement mock unit testing for these methods-- [this related PR](https://github.com/sigstore/rekor-monitor/pull/480) implements some of the rekor client methods necessary to mock test these helper methods. These methods should not be exported, as they are only used for this reusable monitoring workflow.

Additionally, this PR adds an end-to-end test for rekor-monitor into the GitHub PR CI, testing that the reusable monitoring workflow is able to successfully verify the consistency proof of a log tree, as well as successfully monitor for, find, and write identities to file. This end-to-end test is thus able to test the refactored functionality within the rekor-monitor workflow.

Fixes #246 

#### Release Note

NONE

#### Documentation

N/A

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -63,3 +63,30 @@ jobs:
       - name: Run Go tests w/ `-race`
         if: ${{ runner.os == 'Linux' }}
         run: go test -race $(go list ./... | grep -v third_party/)
+  e2e-tests:
```
Comment (mihaimaruseac) [inline .github/workflows/main.yml:66]: Let's separate this job definition from the previous one:

```suggestion


  e2e-tests:
```

**Review by mihaimaruseac — COMMENTED** [review]
LGTM, minor style nits

File: pkg/rekor/verifier.go [hunk 2]
```diff
@@ -40,3 +49,118 @@ func GetLogVerifier(ctx context.Context, rekorClient *client.Rekor) (signature.V
...
+	if err := verify.ProveConsistency(context.Background(), rekorClient, prevCheckpoint, checkpoint, treeID); err != nil {
+		return nil, fmt.Errorf("failed to verify log consistency: %v", err)
+	}
+	fmt.Fprintf(os.Stderr, "Root hash consistency verified - Current Size: %d Root Hash: %s - Previous Size: %d Root Hash %s\n",
+		checkpoint.Size, hex.EncodeToString(checkpoint.Hash), prevCheckpoint.Size, hex.EncodeToString(prevCheckpoint.Hash))
+	return prevCheckpoint, nil
+}
+
+// RunConsistencyCheck periodically verifies the root hash consistency of a Rekor log.
+func RunConsistencyCheck(interval *time.Duration, rekorClient *client.Rekor, verifier signature.Verifier, logInfoFile *string, mvs identity.MonitoredValues, outputIdentitiesFile *string, once *bool) error {
```
Comment (haydentherapper) [inline pkg/rekor/verifier.go:86]: We can do this as a follow up to keep the PR changes smaller (can track with a TODO), but I'd have this take in string and bool values rather than pointers. The code calling `RunConsistencyCheck` should check that the various flag values are set. This also makes it simpler to test as you can assume the value is always set rather than it maybe being null and crashing on dereference.

**Review by haydentherapper — COMMENTED** [review]
You can add "Fixes #246" to the PR description to auto-close the issue once this is merged.

**Review by mihaimaruseac — DISMISSED** [review]

**Review by haydentherapper — APPROVED** [review]
Looks great!
